src/main/java/io/jenkins/plugins/artifact_manager_jclouds/s3/S3BlobStoreConfig.java

```diff
@@ -145,13 +166,78 @@ private void checkValue(@NonNull FormValidation formValidation) {
     }
 
     public boolean isDeleteArtifacts() {
-        return DELETE_ARTIFACTS;
+        return deleteArtifacts;
     }
 
     public boolean isDeleteStashes() {
-        return DELETE_STASHES;
+        return deleteStashes;
+    }
+    
+    public boolean getUsePathStyleUrl() {
+        return usePathStyleUrl;
+    }
+    
+    @DataBoundSetter
+    public void setUsePathStyleUrl(boolean usePathStyleUrl){
+        this.usePathStyleUrl = usePathStyleUrl;
+        save();
+    }
+    
+    public boolean getUseHttp() {
+        return useHttp;
+    }
+
+    @DataBoundSetter
+    public void setUseHttp(boolean useHttp){
+        this.useHttp = useHttp;
+        save();
+    }
+    
+    public boolean getDisableSessionToken() {
+        return disableSessionToken;
     }
 
+    @DataBoundSetter
+    public void setDisableSessionToken(boolean disableSessionToken){
+        this.disableSessionToken = disableSessionToken;
+        save();
+    }
+    
+    public String getCustomEndpoint() {
+        return customEndpoint;
+    }
+    
+    @DataBoundSetter
+    public void setCustomEndpoint(String customEndpoint){
+        this.customEndpoint = customEndpoint;
+        checkValue(doCheckCustomEndPoint(this.customEndpoint));
```

Comment:
This can throw a `Failure` and then `save()` is not run, is that the behavior we want?

Reply:
Yeah it should do the check, then save.